.buildkite/scripts/skip_if_only_docs_changes.sh

```diff
@@ -0,0 +1,17 @@
+# Source this file rather than fork-exec-ing it.
+
+only_docs_changes() {
```

Comment:
I would generalize and rename this to `only_non_code_changes()` (also the `.sh` file) and include other pathspec patterns:
- `:(exclude).github/`
- `:(exclude)docker/`
- `:(exclude)docs/` (in case there will be non-Markdown files in the `docs/` directory, e.g. configuration files, images, etc.
- `:(exclude).codecov.yml`
- `:(exclude).dockerignore`
- `:(exclude).gitignore`
- `:(exclude).gitlint`
- `:(exclude).gitmodules`
- `:(exclude).goreleaser.yml`
- `:(exclude).markdownlint.yml`
- `:(exclude).punch_config.py`
- `:(exclude)towncrier.toml`

Reply:
ok

Reply:
I'll keep `docker/` as code for now

Reply:
hm some of these other ones we'll still want to run CI on:

- codecov, so that we generate coverage for it
- .github/workflows, because ugh this is philosophically part of CI, and we're not clear that we'll never do code builds on github actions, or that this helper is specific to buildkite
- .gitmodules

Reply:
okay overall I'm not ready to extend this to those other non-documentation non-code files

Reply:
I'll scope down this PR

Reply:
> I'll keep `docker/` as code for now

Ok

> codecov, so that we generate coverage for it

Ok

> .github/workflows, because ugh this is philosophically part of CI, and we're not clear that we'll never do code builds on github actions, or that this helper is specific to buildkite

I would keep excluding this one since the helper being in ` .buildkite/scripts/skip_if_only_docs_changes.sh` makes it Buildkite-specific (for now).

> .gitmodules

We should just delete it since it is empty.

> I'll scope down this PR

Ok, this is a good start and we can extend it later.

Reply:
ok